### PR TITLE
Jason/issue 1130

### PR DESCRIPTION
**Pull request by duidae**
companion PR for https://github.com/CARTAvis/carta-frontend/issues/1130

### CODE REVIEW

**Comment by veggiesaurus** [comment]
Can we not do this alive check on the front-end asynchronously? The current approaches blocks the backend while it waits for a response.

**Comment by duidae** [comment]
> Can we not do this alive check on the front-end asynchronously? The current approaches blocks the backend while it waits for a response.

due to CORS we still need backend to interact with splatalogue, and a 20 sec timeout is added to the ping msg.

**Review by kswang1029 — APPROVED** [review]

**Review by veggiesaurus — APPROVED** [review]
looks ok to me. Just waiting for protocol buffer PR reviews then we can merge. @ajm-asiaa not sure why the ICD tests are failing here :thinking: 

edit: oops, spotted some minor issues

File: src/Session.cc [hunk 1]
```diff
@@ -1188,6 +1188,12 @@ void Session::OnSaveFile(const CARTA::SaveFile& save_file, uint32_t request_id)
     }
 }
 
+void Session::OnSplataloguePing(CARTA::SplataloguePing splatalogue_ping, uint32_t request_id) {
```
Comment (veggiesaurus) [inline src/Session.cc]: `splatalogue_ping` is unused, we don't need to pass it here.

File: src/OnMessageTask.h [hunk 2]
```diff
@@ -146,6 +146,19 @@ class SpectralProfileTask : public OnMessageTask {
     ~SpectralProfileTask() = default;
 };
 
+class OnSplataloguePingTask : public OnMessageTask {
+    tbb::task* execute() override;
+    CARTA::SplataloguePing _message;
+    uint32_t _request_id;
+
+public:
+    OnSplataloguePingTask(Session* session, CARTA::SplataloguePing message, uint32_t request_id) : OnMessageTask(session) {
```
Comment (veggiesaurus) [inline src/OnMessageTask.h]: `_message` is unused

File: src/OnMessageTask.cc [hunk 3]
```diff
@@ -145,6 +145,11 @@ tbb::task* SpectralProfileTask::execute() {
     return nullptr;
 }
 
+tbb::task* OnSplataloguePingTask::execute() {
+    _session->OnSplataloguePing(_message, _request_id);
```
Comment (veggiesaurus) [inline src/OnMessageTask.cc]: only need `_request_id`

**Comment by ajm-ska** [comment]
> looks ok to me. Just waiting for protocol buffer PR reviews then we can merge. @ajm-asiaa not sure why the ICD tests are failing here 🤔

It is the segmentation fault of carta_backend_tests on MacOS. I still haven't figured out the reason yet...

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**